pulser/sequence.py

```diff
@@ -217,7 +236,10 @@ def current_phase_ref(self, qubit, basis='digital'):
 
         return self._phase_ref[basis][qubit].last_phase
 
-    def declare_channel(self, name, channel_id, initial_target=None):
+    def declare_channel(self, name: str, channel_id: str,
+                        initial_target: Optional[Union[set[QubitId],
+                                                       QubitId]] = None
```

Comment:
I know it says so in the docstring (you should change it too btw), but the `initial_target` doesn't actually have to be a `set[QubitId]`, but rather `Iterable[QubitId]` (same as the `target` method).

Also, each element can be a variable instead of `QubitId`, so the complete form would be something like:
```suggestion
                        initial_target: Optional[Union[Iterable[Union[QubitId, Parametrized]],
                                                       Union[QubitId, Parametrized]]] = None
```
(perhaps it's a good idea to use an alias, as this seems quite verbose)